### PR TITLE
prevent volt loading on files and hidden file/folders

### PR DESCRIPTION
**Pull request by nheuillet**
- [x] Added an entry to `CHANGELOG.md` if this change could be valuable to users

### CODE REVIEW

File: lapce-proxy/src/plugin/wasi.rs [hunk 1]
```diff
@@ -233,7 +233,14 @@ pub fn find_all_volts() -> Vec<VoltMetadata> {
...
                 dir.filter_map(|result| {
                     let entry = result.ok()?;
+                    if entry.file_type().ok()?.is_file() {
+                        return None;
+                    }
+                    if entry.file_name().to_str()?.starts_with(".") {
+                        return None;
+                    }
                     let path = entry.path().join("volt.toml");
+                    println!("PATH: {:?}", &path);
```
Comment (panekj) [inline lapce-proxy/src/plugin/wasi.rs:243]:
```suggestion
```

**Review by panekj — CHANGES_REQUESTED** [review]

File: lapce-proxy/src/plugin/wasi.rs [hunk 2]
```diff
@@ -233,6 +233,12 @@ pub fn find_all_volts() -> Vec<VoltMetadata> {
             d.read_dir().ok().map(|dir| {
                 dir.filter_map(|result| {
                     let entry = result.ok()?;
+                    if entry.file_type().ok()?.is_file() {
+                        return None;
+                    }
+                    if entry.file_name().to_str()?.starts_with('.') {
+                        return None;
+                    }
```
Comment (panekj) [inline lapce-proxy/src/plugin/wasi.rs:241]:
```suggestion
                    let entry = entry.metadata()?;
                    if !entry.is_dir() || entry.file_name().to_str()?.starts_with('.') {
                        return None;
                    }
```

**Review by panekj — CHANGES_REQUESTED** [review]

File: lapce-proxy/src/plugin/wasi.rs (hunk 2 above)
Comment (nheuillet) [reply lapce-proxy/src/plugin/wasi.rs]: metadata does not contain file_name. I can remove the variable init if you prefer, seeing as it's only used on is_dir

Comment (panekj) [reply lapce-proxy/src/plugin/wasi.rs]: you committed outdated suggestion

Comment (nheuillet) [reply lapce-proxy/src/plugin/wasi.rs]: 🤦 Sorry about that

Comment (panekj) [reply lapce-proxy/src/plugin/wasi.rs]: could probably use `if metadata.is_file()` to not break symlinks

**Review by panekj — APPROVED** [review]

**Pull request merged**